pkg/apiserver/registry/datapackaging/package_storage_client.go

```diff
@@ -152,7 +153,30 @@ func (t PackageTranslator) ToExternalWatcher(intObjWatcher watch.Interface, fiel
 }
 
 func (t PackageTranslator) ToExternalError(err error) error {
-	// TODO: implement
+
+	status, ok := err.(apierrors.APIStatus)
+	if !(ok || errors.As(err, &status)) {
+		return err
+	}
+
+	details := status.Status().Details
+	if details != nil && details.Kind == "internalpackages" && details.Group == internalpkgingv1alpha1.SchemeGroupVersion.Group {
+		packageName, translateErr := t.ToExternalName(details.Name)
+		if translateErr != nil {
+			return err
```

Comment:
We are hiding here a possible translation error. In what situation does this error happen?
Maybe it would make sense to save it somehow, or by wrapping the current error and extending with more information or to log this error out somewhere.

Reply:
ToExternalName decodes the base 64 encoded package name.
```
func (t PackageTranslator) ToExternalName(name string) (string, error) {
	decodedBytes, err := base32.HexEncoding.WithPadding(base32.NoPadding).DecodeString(strings.ToUpper(name))
	return string(decodedBytes), err
}
```

Reply:
Ok, but we are hiding an error, is that ok?

Reply:
Good catch, I somehow overlooked this after my last comment on returning the original error.
I think we should log the error, wrapping the existing error might confuse the users (as the original itself is confusing because of the hashed name)

Reply:
which means we should return the translate error?

Reply:
We should log the translation error.

Reply:
Logger is added to print the translate error.